netshare/generate/model_generate.py

```diff
@@ -72,19 +73,25 @@ def generate_data(
 @ray.remote(scheduling_strategy="SPREAD", max_calls=1)
 def merge_attr(config_group: dict, configs: List[dict]) -> None:
     """
-    TODO: Can someone help me to doc this function?
+    This is the function to merge the attributes. The models trained on different
+    chunks of data will first generate attributes. Then, this function will be
+    invoked to merge the synthetic attributes. Given the merged attributes, models
+    will continue generating the time-series measurements.
+
+    :return: None
     """
     chunk0_idx = config_group["config_ids"][0]
     eval_root_folder = configs[chunk0_idx]["eval_root_folder"]
     attr_raw_npz_folder = os.path.join(eval_root_folder, "attr_raw")
-    word2vec_size = configs[chunk0_idx].get("word2vec_vecSize", 0)
-    pcap_interarrival = configs[chunk0_idx].get("timestamp") == "interarrival"
     num_chunks = len(config_group["config_ids"])
 
-    if not pcap_interarrival:
-        bit_idx_flagstart = 128 + word2vec_size * 3
-    else:
-        bit_idx_flagstart = 128 + word2vec_size * 3 + 1
+    dim = 0
+    for field in list(learn_api.get_attributes_fields().values()):
+        if field.name != "startFromThisChunk":
+            dim += field.getOutputDim()
+        else:
+            break
+    col_idx_of_session_start = dim
```

Comment:
your logic depends on the order of the list, and the list is stemmed from a `dict`'s values.
Generally, we shouldn't do that because `dict` is an unordered data structure.
Maybe you don't really need the `break`?

Reply:
If it works for you, then we can keep it like that, and I'll change it after the big merge !

Reply:
I agree. `List` or `OrderedDict` is definitely better. I also have this concern and when I'm iterating the `dict`, I find it out the order is correct. So I just plan to put it there and discuss it with you. I agree that using list is better, I will make the change on the code. Thanks!!

Reply:
Great, thank you 🙏 
If you prefer, you can leave it like that and I'll fix it in a followup PR

Reply:
Sure. I will leave it to you.